x-pack/test/cases_api_integration/common/lib/utils.ts

```diff
@@ -1347,3 +1350,45 @@ export const getReferenceFromEsResponse = (
   esResponse: TransportResult<GetResponse<SavedObjectsRawDocSource>, unknown>,
   id: string
 ) => esResponse.body._source?.references?.find((r) => r.id === id);
+
+export const suggestUserProfiles = async ({
+  supertest,
+  req,
+  expectedHttpCode = 200,
+  auth = { user: superUser, space: null },
+}: {
+  supertest: SuperTest.SuperTest<SuperTest.Test>;
+  req: SuggestUserProfilesRequest;
+  expectedHttpCode?: number;
+  auth?: { user: User; space: string | null };
+}): ReturnType<UserProfileService['suggest']> => {
+  const { body: profiles } = await supertest
+    .post(`${getSpaceUrlPrefix(auth.space)}${INTERNAL_SUGGEST_USER_PROFILES_URL}`)
+    .auth(auth.user.username, auth.user.password)
+    .set('kbn-xsrf', 'true')
+    .send(req)
+    .expect(expectedHttpCode);
+
+  return profiles;
+};
+
+export const loginUsers = async ({
+  supertest,
+  users = [superUser],
+}: {
+  supertest: SuperTest.SuperTest<SuperTest.Test>;
+  users?: User[];
+}) => {
+  for (const user of users) {
```

Comment:
This can't be done in a `Promise.all` because the order needs to be consistent for the results returned from the suggest API.